src/Softeq.NetKit.Chat.SignalR/Hubs/Notifications/MessageNotificationService.cs

```diff
@@ -29,25 +29,27 @@ public class MessageNotificationService : BaseNotificationService, IMessageNotif
             _channelService = channelService;
         }
 
-        public async Task OnAddMessage(MessageResponse message, string callerConnectionId)
+        public async Task OnAddMessage(MessageResponse message, string callerConnectionId = null)
         {
             var clientIds = await GetChannelClientConnectionIdsAsync(message.ChannelId);
-            var clientIdsExceptCaller = clientIds.Except(new[] { callerConnectionId }).ToList();
+            var clientIdsExceptCaller = callerConnectionId != null
+                            ? clientIds.Except(new[] { callerConnectionId }).ToList()
+                            : clientIds;
 
             await HubContext.Clients.Clients(clientIdsExceptCaller).SendAsync(HubEvents.MessageAdded, message);
         }
 
         public async Task OnDeleteMessage(ChannelSummaryResponse channelSummary, MessageResponse message)
         {
             var clientIds = await GetChannelClientConnectionIdsAsync(message.ChannelId);
-            
+
             await HubContext.Clients.Clients(clientIds).SendAsync(HubEvents.MessageDeleted, message.Id, channelSummary);
         }
 
         public async Task OnUpdateMessage(MessageResponse message)
```

Comment:
We should disable editing for system message.

Reply:
done